nuxeo-features/rest-api/nuxeo-rest-api-server/src/main/java/org/nuxeo/ecm/restapi/server/jaxrs/OAuth2Object.java

```diff
@@ -191,7 +197,23 @@ public Response getToken(@PathParam("providerId") String providerId, @Context Ht
     @Path("token")
     public List<NuxeoOAuth2Token> getTokens(@Context HttpServletRequest request) {
         checkPermission(null);
-        return getTokens();
+        return Framework.getService(OAuth2TokenService.class).getTokens();
+    }
+
+    /**
+     * Retrieves all OAuth2 tokens by oAuth2 token type.
+     *
+     * @param type, the value of {@link NuxeoOAuth2TokenType}
+     * @param request the http request
+     * @return if <code>type</code> is {@link NuxeoOAuth2TokenType#AS_PROVIDER} value, then we retrieve tokens that are provided
+     *         by Nuxeo, otherwise those used by Nuxeo to connect to others applications
+     * @since 11.1
+     */
+    @GET
+    @Path("token/{type}")
+    public List<NuxeoOAuth2Token> getTokens(@PathParam("type") String type, @Context HttpServletRequest request) {
+        checkPermission(null);
+        return Framework.getService(OAuth2TokenService.class).getTokens(NuxeoOAuth2TokenType.fromValueType(type));
```

Comment:
What is the difference between this method and `getProviderUserTokens` or `getClientUserTokens`?

Reply:
my idea is to have just one method and to get token provided or consumed by passing the param. and the new method use a service that filter in the query instead of filtering java side.

to deprecate the old ones i dont think is a good idea instead but is think i will remove the new one and use the  service to retrieve data instead doing it in the resource